oak/server/rust/oak_runtime/src/io.rs

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{runtime::RuntimeProxy, NodeMessage};
+use log::error;
+use oak_abi::{ChannelReadStatus, Handle, OakStatus};
+
+/// A trait for objects that can be decoded from bytes + handles.
```

Comment:
Could you mention that this is the Runtime equivalent of the one in the SDK (and add a link to it or to its documentation). Same for anything else that is copied / inspired by other existing code.